ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosDataAccessor.java

```diff
@@ -155,6 +157,28 @@ void testConnectivity() {
     }
   }
 
+  /**
+   * Get the number of blobs in the specified partition matching the specified DocumentDB query.
+   * @param partitionPath the partition to query.
+   * @param querySpec the DocumentDB query to execute.
+   * @param timer the {@link Timer} to use to record query time (excluding waiting).
+   * @return the number of matching blobs.
+   */
+  int countMetadata(String partitionPath, SqlQuerySpec querySpec, Timer timer)
+      throws DocumentClientException {
+    FeedOptions feedOptions = new FeedOptions();
+    feedOptions.setPartitionKey(new PartitionKey(partitionPath));
+    try {
+      FeedResponse<Document> response = executeCosmosQuery(querySpec, feedOptions, timer).single();
+      return ((Number) response.getResults().get(0).getHashMap().values().stream().findFirst().get()).intValue();
```

Comment:
what is the format of results here? Instead of findFirst could you get the relevant entry by looking up its key?

Reply:
The returned document is: {"_aggregate":20}
I can change it to lookup by the key, it just involves ugly hardcoding and I could break if they decide to change the key name.

Reply:
Never mind, Cosmos does have a constant for it: Constants.Properties.AGGREGATE, so I'll make the change.